llvm/lib/Target/DirectX/DXILDataScalarization.cpp

```diff
@@ -65,11 +67,17 @@ class DataScalarizerVisitor : public InstVisitor<DataScalarizerVisitor, bool> {
 private:
   GlobalVariable *lookupReplacementGlobal(Value *CurrOperand);
   DenseMap<GlobalVariable *, GlobalVariable *> GlobalMap;
+  static bool isArrayOfVectors(Type *T);
```

Comment:
I don't see why this is static?

Reply:
Historical reasons. It started as a helper. similar to `replaceVectorWithArray` but  then I ended up only using it  in the visitor pattern so moved it into the class.  It doesn't need to be static anymore.

Reply:
I don't have a great reason. I tend to make things static when I want to ensure helpers remain stateless. It doesn't need access to the this pointer and I don't  want to allow method overriding. The only thing I thought might be useful was the private visibility modifier.

Reply:
Does it need to be a member function of `DataScalarizerVisitor` at all? I think it'd be clearer to just make this a static freestanding function.

Reply:
No it doesn't.  I can make it free standing.